report/sections/system-design.tex

```diff
@@ -34,10 +35,205 @@ \subsection{Model}
+\end{equation}
+$\theta$ will always be in $(-\pi,\pi]$ span.\\
+
+Due to the physical limits the servos have, two saturation blocks are required.
+These blocks are shown in figure(\ref{fig:PID1}) and the assumption for their limits are (-10,10) $[cm/sec]$. 
+This assumption is done to make the simulation more alike the real application.\\
+\begin{figure}[ht]
+\centering
+\includegraphics[width=0.4\textwidth]{sections/assets/Theta_pi_stepresponce.png}
+\caption{The step response for $\theta$ where the proportional gain $K_p = 4$ and the reference value is $\pi$.}
+\label{fig:Theta_Step}
+\end{figure}
+
+The result of the simulation, the step response for the angular position control system is presented in figure(\ref{fig:Theta_Step}).
+As it shown this is a stable system, it has no steady state error, no oscillation and relative fast rise-time.\\
+
+In this part of the mobile platforms movement it is important to be able to control the spacial position $(x, y)$. 
+As shown in the kinematic equations Eq.(\ref{eq:MotionEq}) the positions are dependent on both the angular velocity $\omega$ and the average velocity $v$. 
+A design of a controller for the average velocity is required to make it possible to control the spacial position. 
+Take note of the two input $AverageVelocity$, $Desired\_Theta$ and the two output $X$, $Y$ in figure(\ref{fig:PID1}).\\
+
+By redefining the spatial error from a Cartesian coordinate system to a polar coordinate system it is possible to design a controller for average velocity.\\
+In this case the error will be defined as following:
+\begin{eqnarray}
+\begin{aligned}
+\Delta x &= Desired\_X - X\\
+\Delta y &= Desired\_Y - Y\\
+Distance &= \sqrt{\Delta x^2 + \Delta y^2}\\
+Desired\_\theta &= atan2(\Delta y, \Delta x)
+\end{aligned}
+\label{eq:cart2polar}
+\end{eqnarray}
+\\
+
+Now by defining the distance as an error to a PID controller for the average velocity, it makes it possible for the mobile platform to stop at the desired position. 
+In this case a P controller, a controller which has only a proportional gain, is enough.\\
+\begin{figure*}[ht]
+\centering
+\includegraphics[width=\textwidth]{sections/assets/V_PID.PNG}
+\caption{Overview of the complete systems block diagram for the spatial position control. figure(\ref{fig:PID1}) is an overview of the "Steering" block and Eq.(\ref{eq:cart2polar}) is the mathematical definition for the block function "Cartesian- to Polar-coordinates Converter".}
+\label{fig:V_PID}
+\end{figure*}
+ 
+Here it comes some of the results for the spacial position control simulation. In this case the desired position is (100, 100), the proportional gain for the angular velocity controller is $K_p = 4$ and the proportional gain for the average velocity controller is $K_p = 5$. If you are interested to see more plots, you can go to the Git-repository under simulation/Diff\_Drive/read\_me and you will find out how to generate more relative plots for this system.\\
```

Comment:
* "Here it comes some" -> "Here comes some".
* `(100, 100)` -> `$(100, 100)$` if we want to be consisent with how we typeset elements (we do).
* The repository path should be specified in `\texttt{...}`.